### PR TITLE
🎨 Change format rules

### PR DESCRIPTION
**Pull request by NeunEinser**
Everything up for discussion

- Added `object-shorthand` eslint rule
- Changed `operatorPosition` to `nextLine`
- Changed line width from 80 to 100
- Reformatted project using `preferSingleLine: true` temporarily, to illustrate how it might look like with the new line length (this rule is experimental, makes formatting significantly slower, and crashes when trying to format `java-edition/src/mcfunction/tree/patch.ts`, so it definitley shouldn't be enabled by default, and I like the idea of leaving the option to the author to potentially wrap lines more than neccessary)

This is meant as an illustration of how the formatting could be improved. I think bumping the line width a bit like this is a good compromise, I tried `120` initially, but that seemed a bit too much in some cases.

### CODE REVIEW

File: packages/core/src/common/externals/NodeJsExternals.ts [hunk 1]
```diff
@@ -169,12 +152,13 @@ class ChokidarWatcherWrapper extends EventEmitter implements FsWatcher {
...
-			.on('change', (path) => this.emit('change', uriFromPath(path)))
-			.on('unlink', (path) => this.emit('unlink', uriFromPath(path)))
-			.on('error', (e) => this.emit('error', e))
+		this.#watcher = watcher.on('ready', () => this.emit('ready')).on(
+			'add',
+			(path) => this.emit('add', uriFromPath(path)),
+		).on('change', (path) => this.emit('change', uriFromPath(path))).on(
+			'unlink',
+			(path) => this.emit('unlink', uriFromPath(path)),
+		).on('error', (e) => this.emit('error', e))
```
Comment (SPGoding) [inline packages/core/src/common/externals/NodeJsExternals.ts:161]: `preferSingleLine: true` definitely ruined this lol but in general I'm fine with these changes

**Review by SPGoding — APPROVED** [review]

File: packages/core/src/common/externals/NodeJsExternals.ts (hunk 1 above)
Comment (NeunEinser) [reply packages/core/src/common/externals/NodeJsExternals.ts:161]: _I can change it to 99 lmao_

Comment (NeunEinser) [reply packages/core/src/common/externals/NodeJsExternals.ts]: I fixed the formatting of this file manually.

Comment (TheAfroOfDoom) [reply packages/core/src/common/externals/NodeJsExternals.ts]: 99 feels weird, id be happier with 90 or even 95 as a rounder number. but /shrug

**Review by misode — APPROVED** [review]
Looks pretty good. There are some places where the formatter has made it a bit worse, but I'm sure we can cope with that. I wouldn't increase the column limit anymore, but I agree that 80 is maybe a bit too conservative.

**Comment by Calverin** [comment]
I think the length change is a big improvement! I've noticed some areas where the length cutoff seemed a bit much so I'll be happy to see this implemented.

**Pull request merged**